app/assets/javascripts/browsered/web-payments/google-pay.js

```diff
@@ -39,6 +44,95 @@ const processPayment = paymentData => {
+      const { MessageType, SessionId, Status } = JSON.parse(event.data)
+      if (MessageType === 'profile.completed') {
+        var ddcStatus = null
+        var ddcResult = null
+        if (Status === true && typeof SessionId === 'string') {
+          ddcStatus = 'valid DDC result'
+          ddcResult = SessionId
+        } else if (!Status) {
+          ddcStatus = 'DDC result did not have Status of true'
+        } else {
+          ddcStatus = 'no SessionID string in DDC result'
+        }
+        Charge.googlePayWorldpay3dsFlexDeviceDataCollectionResult = ddcResult
+        window.clearTimeout(deviceDataCollectionTimeout)
+        window.removeEventListener('message', deviceDataCollectionFinishedEventListener)
+        submitGooglePayAuthRequest(paymentData, ddcStatus, ddcResult)
+      }
+    }
+  }
+
+  const DDC_TIMEOUT_IN_MILLISECONDS = 30000
+
+  const deviceDataCollectionTimeout = window.setTimeout(() => {
+    window.removeEventListener('message', deviceDataCollectionFinishedEventListener)
+    submitGooglePayAuthRequest(paymentData, `DDC timeout after ${DDC_TIMEOUT_IN_MILLISECONDS} milliseconds`)
+  }, DDC_TIMEOUT_IN_MILLISECONDS)
+
+  var iframe = document.getElementById('googlePayWorldpay3dsFlexDdcIframe')
+
+  try {
+    // eslint-disable-next-line no-unused-expressions
+    iframe.contentWindow.location.href
+  } catch (error) {
+    var iframeParent = iframe.parentElement
+    iframe.remove()
+    var newIframe = document.createElement('iframe')
+    newIframe.id = 'googlePayWorldpay3dsFlexDdcIframe'
+    newIframe.classList.add('govuk-!-display-none')
+    newIframe.src = '/public/worldpay/worldpay-3ds-flex-ddc.html'
+    iframeParent.appendChild(newIframe)
+    iframe = newIframe
+  }
+
+  const iframeWindow = iframe.contentWindow
+  const iframeDocument = iframeWindow.document
+
+  window.addEventListener('message', deviceDataCollectionFinishedEventListener)
+
+  const initiateDeviceDataCollection = iframeDocument => {
+    const innerForm = iframeDocument.getElementById('collectionForm')
+    innerForm.action = Charge.worldpay_3ds_flex_ddc_url
+    iframeDocument.getElementById('input-jwt').value = Charge.worldpay_3ds_flex_ddc_jwt
+    innerForm.submit()
+  }
+
+  if (iframeDocument.readyState === 'complete') {
+    initiateDeviceDataCollection(iframeDocument)
+  } else {
+    iframeDocument.addEventListener('readystatechange', function () {
+      if (iframeDocument.readyState === 'complete') {
+        initiateDeviceDataCollection(iframeDocument)
+      }
+    }, false)
+  }
+}
+
+const processPayment = paymentData => {
+  toggleSubmitButtons()
+  showSpinnerAndHideMainContent()
+
+  if (typeof Charge.worldpay_3ds_flex_ddc_jwt === 'string' && Charge.worldpay_3ds_flex_ddc_jwt !== '') {
+    const cachedDdcResult = window.Charge.googlePayWorldpay3dsFlexDeviceDataCollectionResult
+    if (!cachedDdcResult || !(typeof googlePayDDCEnabled === 'string')) {
```

Comment:
Looks like `googlePayDDCEnabled` is not defined anywhere and is not needed?

Reply:
👍